frontend/server/src/main/java/org/pytorch/serve/metrics/MetricCollector.java

```diff
@@ -40,23 +42,37 @@ public void run() {
             String pythonEnv;
             if ((pythonPath == null || pythonPath.isEmpty())
                     && (!workingDir.getAbsolutePath().contains("site-package"))) {
-                pythonEnv = "PYTHONPATH=" + workingDir.getAbsolutePath();
+                pythonEnv = workingDir.getAbsolutePath();
             } else {
-                pythonEnv = "PYTHONPATH=" + pythonPath;
+                pythonEnv = pythonPath;
                 if (!workingDir.getAbsolutePath().contains("site-package")) {
                     pythonEnv += File.pathSeparatorChar + workingDir.getAbsolutePath(); // NOPMD
                 }
             }
             // sbin added for macs for python sysctl pythonpath
+            HashMap<String, String> environment = new HashMap<>(System.getenv());
+            environment.put("PYTHONPATH", pythonEnv);
+
             StringBuilder path = new StringBuilder();
-            path.append("PATH=").append(System.getenv("PATH"));
+            path.append(System.getenv("PATH"));
             String osName = System.getProperty("os.name");
             if (osName.startsWith("Mac OS X")) {
                 path.append(File.pathSeparatorChar).append("/sbin/");
             }
-            String[] env = {pythonEnv, path.toString()};
-            final Process p = Runtime.getRuntime().exec(args, env, workingDir);
 
+            environment.put("PATH", path.toString());
+            ArrayList<String> envList = new ArrayList<>();
+            Pattern blackList = configManager.getBlacklistPattern();
+
+            for (Map.Entry<String, String> entry : environment.entrySet()) {
```

Comment:
Seems like we are repeating this code. This logic exists in [WorkerLifeCycle.java](https://github.com/pytorch/serve/blob/376110010a2fe3f8341d6299ec60795eaead7bf6/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerLifeCycle.java#L44) as well. We could probably pull this logic of pulling environtment variables and filtering out blacklisted environment variables into a util file and use the methods here and in workerlifecycle.

Reply:
@harshbafna Merging this for now. Created an #386 to refactor this to a utils class